File: 0692-top-k-frequent-words/0692-top-k-frequent-words.cpp
class Solution {
public:
    struct cmp{
    bool operator()(pair<int,string> &a,pair<int,string> &b) const {
        if (a.first != b.first) return a.first < b.first;

            return a.second > b.second;
    }
};

    
    vector<string> topKFrequent(vector<string>& words, int k) {
        vector<string> ans;
        map<string,int> M;
        for(int i=0;i<words.size();i++){
            M[words[i]]++;
        }
        priority_queue<pair<int,string>,vector<pair<int,string>>,cmp> pq;
        for(auto x: M){
            pq.push({x.second,x.first});
        }
        while(k--){
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};

File: 0692-top-k-frequent-words/0692-top-k-frequent-words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        cnt = {}
        for word in words: 
            cnt[word] = cnt.get(word, 0) + 1
        arr = []
        for word,c in cnt.items():
            arr.append([c,word])
        
        arr.sort(key = lambda x : (-x[0],x[1]))
        return [word for _,word in arr[0:k]]

        

        
                
                
                
                

